src/client.cpp

```diff
@@ -1,104 +1,137 @@
 #include "client.hpp"
-
 #include <boost/asio.hpp>
 #include <iostream>
 #include <nlohmann/json.hpp>
 
 using boost::asio::ip::tcp;
 
 namespace meow::network {
-Client::Client() : io_context() {}
+Client::Client() : io_context() {
+}
 
 Client::~Client() = default;
 
+Client &Client::get_instance() {
+    static Client instance;
+    return instance;
+}
+
 void Client::accept_info_about_players() {
-  std::size_t count_of_players;
-  connection >> count_of_players;
-  connection >> id_of_client;
-
-  for (std::size_t i = 0; i < count_of_players; ++i) {
-    std::string input_msg;
-    while (input_msg.empty()) {
-      std::getline(connection, input_msg);
+    std::size_t count_of_players;
+    connection >> count_of_players;
+    connection >> id_of_client;
+
+    for (std::size_t i = 0; i < count_of_players; ++i) {
+        std::string input_msg;
+        while (input_msg.empty()) {
+            std::getline(connection, input_msg);
+        }
+        auto json = json::parse(input_msg);
+        players_info.emplace_back(json);
```

Comment:
I think we need some sort of validation that we received what we were expecting to receive (probably add it to your TODO list)